src/librustc_driver/pretty.rs

```diff
@@ -732,14 +732,18 @@ pub fn pretty_print_input(sess: Session,
     // There is some twisted, god-forsaken tangle of lifetimes here which makes
     // the ordering of stuff super-finicky.
     let mut hir_forest;
-    let lcx = LoweringContext::new(&sess, Some(&krate));
-    let arenas = ty::CtxtArenas::new();
+    let mut _defs = None;
     let dep_graph = DepGraph::new(false);
+    let arenas = ty::CtxtArenas::new();
     let _ignore = dep_graph.in_ignore();
     let ast_map = if compute_ast_map {
+        _defs = Some(RefCell::new(hir_map::collect_definitions(&krate)));
+        let defs = _defs.as_ref().unwrap();
```

Comment:
I'm not sure, but perhaps this could be simplified to:
```rust
let def;
// ...
    defs = RefCell::new(hir_map::collect_definitions(&krate));
    // and then use `&defs` in place of `defs` in the following lines
```